### ISSUE TITLE
`misc-const-correctness.WarnPointersAsValues` not reported

### ISSUE THREAD

**Issue by firewave** [llvm/llvm-project#58098]
```cpp
class A
{
public:
    void f(A*);
};

class C {
private:
    void f();
    A *a();
};

void C::f()
{
    A * /*const*/ a1 = a(); // no warning
    A * const a2 = a();
    a1->f(a2);
}
```
https://godbolt.org/z/cWKG6xWEv

If you remove the function call the warning is reported.

CC @JonasToth

**Comment by felix642** [comment]
Hi @firewave, this doesn't look like a false negative to me.

There is no warning on the variable a1 because you're calling the method f(A*) which is non-const. If you add the const-qualifier to the method than the warning is generated:

https://godbolt.org/z/asdPE79T5

Am I missing something?

**Comment by firewave** [comment]
> There is no warning on the variable a1 because you're calling the method f(A*) which is non-const. If you add the const-qualifier to the method than the warning is generated:

That seems to make no sense at all.

I also filed #64955 which is similar and seems to highlight the issue better. It seems to use the logic for making the type `const` for the check on the pointer.

Playing around a bit with this it seems there might a few more shortcomings here with having the wires of the different logics crossed. I guess that needs more testing and a closer look when someone decides to work on this.
